source/03-components/button/_button.scss

```diff
@@ -18,21 +18,28 @@
   }
 }
 
+// This button class is used for Drupal tabs.
+.button--base {
+  @include button(
+    $color-background: gesso-color(button, base, background),
+    $color-text: gesso-color(button, base, text),
+    $color-background-hover: gesso-color(button, base, background-hover),
+    $color-text-hover: gesso-color(button, base, text-hover),
+    $color-background-active: gesso-color(button, base, background-active),
+    $color-text-active: gesso-color(button, base, text-active)
+  );
+}
+
 // Drupal outputs this class on buttons that can delete content.
 .button--danger {
-  background-color: gesso-color(button, danger, background);
-  color: gesso-color(button, danger, text);
-
-  &:hover,
-  &:focus {
-    background-color: gesso-color(button, danger, background-hover);
-    color: gesso-color(button, danger, text-hover);
-  }
-
-  &:active {
-    background-color: gesso-color(button, danger, background-active);
-    color: gesso-color(button, danger, text-active);
-  }
+  @include button(
+    $color-background: gesso-color(button, danger, background),
+    $color-text: gesso-color(button, danger, text),
+    $color-background-hover: gesso-color(button, danger, background-hover),
+    $color-text-hover: gesso-color(button, danger, text-hover),
+    $color-background-active: gesso-color(button, danger, background-active),
+    $color-text-active: gesso-color(button, danger, text-active)
+  );
```

Comment:
There's a lot of repeated styles here, because button--danger also gets the button class. If your thought is to use the button mixin for all button variants, then it either shouldn't be included on the base `.button` or button variants shouldn't also get the `.button` class. This may also require some updates to the button field formatter to make sure it outputs the right combo of classes for button styles.

Reply:
Hmm, good point. I’ll change the variants to not use the button mixin.

Reply:
Good point. I changed the variants to not use the button mixin.